db/write_thread.cc

```diff
@@ -269,8 +269,8 @@ void WriteThread::CompleteFollower(Writer* w, WriteGroup& write_group) {
   SetState(w, STATE_COMPLETED);
 }
 
+static WriteThread::AdaptationContext jbg_ctx("JoinBatchGroup");
```

Comment:
Yeah i guess that work too. I up for either approach. I guess we wanted them local because they are tied very much to the function. As far as I can a shipit I am ok either way.